File: data/avro-schema/src/main/java/net/corda/data/CordaAvroSerializationFactory.java
package net.corda.data;

import org.jetbrains.annotations.NotNull;

import java.util.function.Consumer;

/**
 * Defines the interface for Message Bus deserialization.  The underlying mechanism may differ.
 */
public interface CordaAvroSerializationFactory {
    /**
     * Create the {@link CordaAvroSerializer} for use in Avro/Message bus serialization
     */
    @NotNull
    <T> CordaAvroSerializer<T> createAvroSerializer(@NotNull Consumer<byte[]> onError);

    /**
     * Create the {@link CordaAvroDeserializer} for use in Avro/Message bus serialization
     */
    @NotNull
    <T> CordaAvroDeserializer<T> createAvroDeserializer(
        @NotNull Consumer<byte[]> onError,
        @NotNull Class<T> expectedClass
    );
}

File: data/avro-schema/src/main/kotlin/net/corda/data/CordaAvroSerializationFactory.kt
package net.corda.data

import java.util.function.Consumer

/**
 * Defines the interface for Message Bus deserialization.  The underlying mechanism may differ.
 */
interface CordaAvroSerializationFactory {
    /**
     * Create the [CordaAvroSerializer] for use in Avro/Message bus serialization
     */
    fun <T : Any> createAvroSerializer(onError: Consumer<ByteArray>): CordaAvroSerializer<T>

    /**
     * Create the [CordaAvroDeserializer] for use in Avro/Message bus serialization
     */
    fun <T : Any> createAvroDeserializer(
        onError: Consumer<ByteArray>,
        expectedClass: Class<T>
    ): CordaAvroDeserializer<T>
}
